### ISSUE TITLE
Pulling AWS CloudWatch logs from an S3 bucket.

### ISSUE THREAD

**Issue by ghost** [DataDog/datadog-serverless-functions#224]
**Describe what happened:**

At our company, we have a system that uses a Kinesis Data Stream to take logs from a CloudWatch group and dump them into an S3 bucket. It is from this S3 bucket that we want our logs to be sent to Datadog. An example file in our S3 bucket looks like this:

```json
{
  "messageType":"DATA_MESSAGE",
  "owner":"***",
  "logGroup":"OverlapNetwork-VpcFlowLogs-191J53EMBBCRQ-LogGroup-AW0BZ8VQNM3V",
  "logStream":"eni-0b637efff203cbe0b-all",
  "subscriptionFilters":["****"],
  "logEvents": [
{"id":"35322448436314467953163414326907908605993876997091885056","timestamp":1583913368000,"message":"2 unknown eni-0b637efff203cbe0b - - - - - - - 1583913368 1583913475 - NODATA"}
  ]
}
{
 "messageType":"DATA_MESSAGE",
  "owner":"261154102564",
  "logGroup":"/ecs/*****/",
  "logStream":"datadog/datadog/a7e3f0a1-4b8a-49da-af64-a40a2e4ab21e",
  "subscriptionFilters":["****"],"
  logEvents":[
{"id":"35322454655657792645376249155738670693078760903753662464","timestamp":1583913646885,"message":"2020-03-11 08:00:46 UTC | TRACE | INFO | (pkg/trace/info/stats.go:108 in LogStats) | [lang:python lang_version:3.6.10 interpreter:CPython tracer_version:0.34.0] -> traces received: 12, traces filtered: 0, traces amount: 30824 bytes, events extracted: 0, events sampled: 0"}
  ]
}
```

**Describe what you expected:**

We expected that the Datadog Log Collection Lambda in this repo would be able to parse these objects like CloudWatch logs, just as if we had the Datadog Log Collection Lambda be triggered from CloudWatch directly.

**Steps to reproduce the issue:**

1. Save CloudWatch logs to S3.
2. Trigger the Datadog Log Collection Lambda upon new ObjectCreate events in the S3 bucket.
3. View the logs in the Datadog Log console.

Tl:dr; we'd like to use Datadog for log analysis, and S3 (& glacier) for long-term long storage. We thought we could achieve both with this setup. Is there another way you'd recommend?

Thank you,

**Comment by ghost** [comment]
It looks like my issue is a repeat of #151. Supporting that would be awesome.
